tests/CMakeLists.txt

```diff
@@ -45,16 +45,23 @@ add_executable(bpftrace_test
   ${CODEGEN_SRC}
 )
 
-# Regenerate DWARF data for unit tests
-if (LIBDW_FOUND)
-  find_program(XXD xxd REQUIRED)
-  add_custom_target(testing_debuginfo_data
-          COMMAND
-          ${CMAKE_COMMAND}
-          -DDATA_SOURCE_DIR="${CMAKE_SOURCE_DIR}/tests/data"
-          -P ${CMAKE_SOURCE_DIR}/tests/data/generate_debuginfo_data.cmake)
-  add_dependencies(bpftrace_test testing_debuginfo_data)
-endif()
+# Regenerate debuginfo data for unit tests
+find_program(XXD xxd REQUIRED)
+find_program(PAHOLE pahole REQUIRED)
+find_program(LLVM_OBJCOPY
+  NAMES llvm-objcopy llvm-objcopy-${LLVM_VERSION_MAJOR} llvm${LLVM_VERSION_MAJOR}-objcopy
+  REQUIRED)
+find_program(NM nm REQUIRED)
+find_program(AWK awk REQUIRED)
+find_program(STRIP strip REQUIRED)
+add_custom_target(testing_debuginfo_data
+        COMMAND
+        ${CMAKE_COMMAND}
+        -DDATA_SOURCE_DIR="${CMAKE_SOURCE_DIR}/tests/data"
+        -DLIBDW_FOUND="${LIBDW_FOUND}"
+        -DLLVM_OBJCOPY="${LLVM_OBJCOPY}"
+        -P ${CMAKE_SOURCE_DIR}/tests/data/generate_debuginfo_data.cmake)
```

Comment:
Perhaps consider making `generate_debuginfo_data.cmake` define a target and using the target instead? Not sure if source file dependencies are properly being tracked right now.

Looks like we are running `cmake` like a script here. Which is ok, I guess, since it's extending the existing stuff. But converting the `generate_debuginfo_data.cmake` into a shell script or python script may make it more readable.

Reply:
> Perhaps consider making `generate_debuginfo_data.cmake` define a target and using the target instead? Not sure if source file dependencies are properly being tracked right now.

That's a good idea, I'll give it a try.

> Looks like we are running `cmake` like a script here. Which is ok, I guess, since it's extending the existing stuff. But converting the `generate_debuginfo_data.cmake` into a shell script or python script may make it more readable.

The main reason why this is a cmake script is that it uses `configure_file` to create the header files. Bash or Python probably have something similar but CMake's file generation is quite elegant.

Reply:
Ha, so we seem to have had the same discussion already :-) https://github.com/iovisor/bpftrace/pull/1994#discussion_r719811402

I remember trying multiple alternative approaches but none was working. Anyways, this one seems to handle file dependencies properly so I'd keep it as it is.